react/features/base/dialog/components/MessageDialog.web.js

```diff
@@ -0,0 +1,170 @@
+        this._onDontShowChanged = this._onDontShowChanged.bind(this);
+        this._onSubmit = this._onSubmit.bind(this);
+    }
+
+    /**
+     * Load from local storage the option does user had indicated that we should
+     * not show this message dialog.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillMount() {
+        if (this.props.doNotShowWarningLocalStorageKey) {
+            this.setState({
+                doNotShowDialog:
+                    jitsiLocalStorage.getItem(
+                        this.props.doNotShowWarningLocalStorageKey)
+                            === 'true'
+            });
+        } else {
+            this.setState({
+                doNotShowDialog: false
+            });
+        }
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        if (this.state.doNotShowDialog) {
+            return null;
+        }
+
+        const doNotShowWarning = this._renderDoNotShowWarning();
+
+        return (
+            <Dialog
+                onSubmit = { this._onSubmit }
+                { ...this.props }>
+                { this.props.message }
+                { this.props.children }
+                { doNotShowWarning }
+            </Dialog>
+        );
+    }
+
+    /**
+     * Creates a 'do not show dialog' component that contains a toggle which
+     * user can use to indicate that this is the last time he sees this dialog.
+     * If 'do not show dialog' component is not enabled returns null.
+     *
+     * @private
+     * @returns {ReactComponent|null}
+     */
+    _renderDoNotShowWarning() {
+        if (!this.props.doNotShowWarningLocalStorageKey
+            || !this.props.doNotShowWarningTextKey) {
+            return null;
+        }
+
+        return (
+            <h5>
+                <Label
+                    label
+                    = { this.props.t(this.props.doNotShowWarningTextKey) } />
+                <Toggle onChange = { this._onDontShowChanged } />
+            </h5>);
+    }
+
+    /**
+     * Updates state when toggle state changes.
+     *
+     * @param {Object} e - The event coming from Toggle component.
+     * @private
+     * @returns {void}
+     */
+    _onDontShowChanged(e) {
+        this.setState({
+            doNotShowWarningChecked: e.target.checked
+        });
+    }
+
+    /**
+     * Dispatches the action when submitting the dialog. If
+     * 'do not show dialog' component is enabled and its toggle is checked
+     * store the desired value in localstorage in order to dismiss future
+     * dialogs like current one.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onSubmit() {
+        if (this.state.doNotShowWarningChecked) {
+            jitsiLocalStorage.setItem(
+                this.props.doNotShowWarningLocalStorageKey, 'true');
+        }
+
+        return true;
```

Comment:
So the mute participant needs to do more work on checking doNotShowWarning? This is supposed to be a localstorage key, common way of preventing a dialog to show. Maybe mute participant should trigger opening that dialog and the dialog itself will decide whether it will show or not, do not see why it needs any logic specific to mute participant feature. WDYT?